scripts/screeps_tencent_batch_rl_runner.py

```
@@ -2064,6 +2085,165 @@ def verified_remote_policy_update_fields(
+        if injected_count <= 0:
+            raise BatchRunError("remote candidateScorecard ready status requires positive injectedVariantCount")
+        nested_scorecard_id = required_non_empty_text(raw.get("scorecardId"), "candidateScorecard.scorecardId")
+        top_level_scorecard_id = required_non_empty_text(scorecard_id, "scorecardId")
+        if nested_scorecard_id != top_level_scorecard_id:
+            raise BatchRunError("remote candidateScorecard.scorecardId disagrees with scorecardId")
+        safe_candidate_scorecard_artifact_path(scorecard_artifact_path)
+    elif status == "blocked":
+        classification = raw.get("classification")
+        materialization_failed = classification == "candidate_scorecard_materialization_failed"
+        if runtime_injected and not materialization_failed:
+            raise BatchRunError("remote candidateScorecard blocked status requires runtimeParameterInjection=false")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)